### PR TITLE
Code Insights: Add the cloud version of the landing page

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes part of https://github.com/sourcegraph/sourcegraph/issues/30166
Desings [Figma](https://www.figma.com/file/EFgTP7gpdpQSEo1n4FUfsW/In-product-landing-page-WIP?node-id=1791%3A29077)

## Test plan
Just check that UI parts look good comparing to our landing page desings 

## Run this locally
`sg start web-standalone` should be enough. In code insights code we check sourcegraphDotCome mode which is on for the local dev so you should see a cloud landing page by default on any /insights/ routes within the app. Note: you shouldn't see other app-related routes. 

Also go to the `CodeInsightsRouter.tsx` and set `CLOUD_LANDING_PAGE` to true.

## ⚠️  Before merging 
Make sure that `CLOUD_LANDING_PAGE` is equal to `false` to avoid releasing "work in progress" versions of the landing page.

## Todo 
- [x] Add the main layout for the landing page
- [x] Add the first blank section 
- [x] Add quote section 
- [x] Add layout for the examples section 
- [x] Add picker logic for the examples section and reuse chart example components from the in-product landing page
- [x] Add requirements and "install locally" sections/cards layouts
- [x] Reuse template section from the in-product landing page
- [x] Add footer with links layout

## Important parts but it will be implemented in a separate PR https://github.com/sourcegraph/sourcegraph/issues/31650

- [ ] Add promo video 
- [ ] Add gif (hero image) for the first section
- [ ] Add pings
- [ ] Make sure that pings from the in-product landing page and cloud landing page work independently

### CODE REVIEW

**Comment by Joelkw** [comment]
@vovakulikov noting that the cloud landing page shouldn't go live until Feb 24th. We will merge it to cloud then.

**Comment by felixfbecker** [comment]
@vovakulikov would it make sense to make it a global flag instead we can toggle at GA time (so we can merge 99% of the work earlier, and deploy the switch faster)? What are your thoughts?

**Comment by vovakulikov** [comment]
@felixfbecker Agree. 

We could have just a local js variable (just to avoid usage of the setting cascade cause it isn't so private in this case, I mean users probably can find it before the GA date). Close to the GA data we just merge a simple PR where we remove this variable.

File: client/web/src/enterprise/insights/pages/dot-com-get-started/CodeInsightsDotComGetStarted.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                </Card>
+            </section>
+
+            <CodeInsightsTemplates
+                interactive={false}
+                className={styles.templateSection}
+                telemetryService={telemetryService}
+            />
+
+            <section className={styles.videoSection}>Demo vide placeholder</section>
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/pages/dot-com-get-started/CodeInsightsDotComGetStarted.tsx:121]:
```suggestion
            <section className={styles.videoSection}>Demo video placeholder</section>
```

**Review by unclejustin — APPROVED** [review]

**Pull request merged**